Develop

ValeriaSkopinceva opened the pull request.

ValeriaSkopinceva committed:
создан класс person, ветка develop

ValeriaSkopinceva committed:
добавление групп
добавление/удаление студентов в группу/из группы

ValeriaSkopinceva committed:
добавление групп
добавление/удаление студентов в группу/из группы
создание окна в графическом интерфейск

ValeriaSkopinceva committed:
инициализирование классов Lectures, Subject
исправление ошибки в display...(но еще есть проблемы с выводом)
разделение findGroup на два класса( один возвращает Group, другой boolean)

ValeriaSkopinceva committed:
Добавление класса Assembly(сборщик). Нужен для реализации методов, которые будут потом использоваться в GraphicallInterface(нажал на определенную кнопку - вызвался метод из Assembly)
Добавление класса-списка и поля-списка Weekday. Непонятно, зачем нужен класс, если и в поле выполняются нужные действия( но как-то очень странно, слишком много методов)

ValeriaSkopinceva committed:
Создала ActionListener для b1
Создала текстовые поля для ввода информации

ValeriaSkopinceva committed:
Создала ActionListener для b1
Создала текстовые поля для ввода информации
Заполнено расписание для групп
Созданы само группы, из main перенесены в GroupService

andreyoganesyan commented on src/timetable/model/Assembly.java:
```diff
@@ -0,0 +1,29 @@
+import static timetable.model.Weekday.search;
+
+
+public class Assembly {
```
Ни один метод в этом классе не работает. Если вы его не используете, надо его удалить

ValeriaSkopinceva replied:
класс не закончен еще. не придумала еще, как реализовать поиск

andreyoganesyan commented on src/timetable/model/Lesson.java:
```diff
@@ -0,0 +1,94 @@
+
+
+    private void weeklySchedule(){  // заполнение списка предметов и расписания для групп
+        //  МОЖНО ЛИ ЭТО ДЕЛАТЬ НЕ ОТ РУКИ?
```
По-хорошему такие вещи читаются из базы данных или на худой конец из файла. Мы с этим работать не учились, поэтому пусть остается заполнение от руки.

andreyoganesyan commented on src/timetable/model/Lesson.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+
+    private void weeklySchedule(){  // заполнение списка предметов и расписания для групп
```
Инициализация данных — сервисная логика, а не логика модели. Этот метод надо перенести в какой-нибудь сервис

andreyoganesyan commented on src/timetable/model/Group.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.Set;
+
+public class Group {
+    private Set<Student> students;
```
Вы используете Student в Set, но в нем не переопределены equals и hashCode, так делать не стоит: https://javarush.ru/groups/posts/1340-peregruzka-metodov-equals-i-hashcode-v-java

ValeriaSkopinceva replied:
+

ValeriaSkopinceva committed:
Создала ActionListener для b1
Создала текстовые поля для ввода информации
Заполнено расписание для групп
Созданы сами группы, из main перенесены в GroupService

andreyoganesyan commented on src/timetable/model/Weekday.java:
```diff
@@ -0,0 +1,62 @@
+    }
+    public Weekday toDisplayLessonsForDays() {
+        if (MONDAY.equals(this)) {
+            System.out.println("Monday");
```
Никаких System.out.println в модели быть не должно
Не понял смысл этого метода, вы фактически везде возвращаете this. Неужели только для System.out.println он сделан?

ValeriaSkopinceva replied:
остался с прошлого изменения,когда только проверяла работу enum. забыла удалить

andreyoganesyan commented on src/timetable/service/GraphicalInterface.java:
```diff
@@ -0,0 +1,76 @@
+package timetable.service;
```
Графический интерфейс явно должен быть во view, а не в service

andreyoganesyan commented on src/timetable/service/GraphicalInterface.java:
```diff
@@ -0,0 +1,76 @@
+    JLabel l1,l2,l3;
+    String name, surname;
+    int course, group;
+    eHandler handler = new eHandler();
```
Почему не приватные поля?

ValeriaSkopinceva replied:
изменила

andreyoganesyan commented on src/timetable/service/GroupService.java:
```diff
@@ -0,0 +1,90 @@
+        students.remove(studentToRemove);
+    }
+
+    public void enterGroups(ArrayList<Group> groups) { //  вывод списка групп
```
Это view

ValeriaSkopinceva replied:
+

andreyoganesyan commented on src/timetable/service/GroupService.java:
```diff
@@ -0,0 +1,90 @@
+        }
+    }
+
+    public void displayStudentInGroup(ArrayList<Group> groups, int number, int course) {  //вывод студентов в конкретной группе УРАААААААААААААА
```
И это view

ValeriaSkopinceva replied:
+

andreyoganesyan commented on src/timetable/service/GroupService.java:
```diff
@@ -0,0 +1,90 @@
+            System.out.println("Нет такой группы");
+    }
+
+    private void zapol(){
```
Давайте все-таки использовать английские слова в названиях

ValeriaSkopinceva replied:
это временное название

andreyoganesyan commented on src/timetable/view/GroupView.java:
```diff
@@ -0,0 +1,34 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Group {
```
Не должен ли этот класс взаимодействовать с GroupService? В чем тогда его смысл?

ValeriaSkopinceva committed:
Обработчик для кнопки 3 и 4
Разделение по пакетам
Отладка расположения кнопок на окне
Создание LessonService
